Min # Jsonparser step2

undervineg opened the pull request.

undervine committed:
Use different way to parse data

Add extractArrayBlobs(), extractObjectBlobs() to get arrays of object pattern or array pattern data.
Add convertToDictionary(), generateKey(), generateValue() to convert each String data to dictionary.

undervine committed:
Add some preprocesses to askFor() func

Modify askFor() func to remove outermost square brackets of user input String.

undervine committed:
Add DataType enum and type property that stores DataType

DataType enum indicates whether type of data is Array or Object.

undervine committed:
Add if statement to check whether user input contains square bracket

undervine committed:
Add getOverAllReport(), getSpecificReport() func to separate output message

undervine committed:
Add some unit test functions

Add functions to test the result of parsing each data type is not nil.

undervine committed:
Modify if statement where checks secondCharacter is '{' and '[

undervine committed:
Modify getOverallReport() func

Make getOverallReport() to print the number of objects and the number of arrays separately.

undervine committed:
Add unit tests

godrm commented on JSONParser/JSONParser/InputView.swift:
```diff
@@ -9,38 +9,61 @@
+        // 문자열 처리를 쉽게 하기 위해 모든 공백 제거.
+        let inputWithoutWhitespace = inputLine.filter { $0 != " " }
+        // JSON 규격 여부 검사.
+        guard try isJSONPattern(inputWithoutWhitespace) else { throw JSONParser.JsonError.invalidPattern }
```
askFor() 함수에 `사용자 입력값 전처리` 기능을 추가하면서 단일 책임 원칙이 깨진 것 같네요.
함수가 한 가지 동작만 하도록 메소드를 분리해보세요.

undervine committed:
Move functions from InputView to JSONParser that preprocess user input

Separate logic of preprocessing user input to func trimExceptDataInside().
Move trimExceptDataInside() and removeSquareBracket() from InputView struct to JSONParser struct.

undervineg commented:
- 전처리 함수들을 InputView에서 JSONParser 구조체로 이동.
- InputView 구조체에서는 사용자 입력값의 JSON 규격 유효성만 체크.

undervine committed:
Modify files according to codacy report

godrm commented:
코드가 읽기 좋습니다. 다음 단계에 도전하세요 💯
